src/echo/ws-api/wallet-api.js

```diff
@@ -954,70 +853,63 @@ class WalletAPI {
 	 * @returns {Promise<string | undefined>}
 	 */
 	getAccountByAddress(address) {
-		if (!isRipemd160(address)) throw new Error('Address id should be a 20 bytes hex string');
-
-		return this.wsRpc.call([0, 'get_account_by_address', [address]]);
+		return this.wsRpc.call([0, 'get_account_by_address', [ripemd160.toRaw(address)]]);
 	}
 
 	/**
 	 * @method getAccountWithdrawals
-	 * @param {String} accountId
+	 * @param {String} idOfAccount
 	 * @returns {Promise<Object[]>}
 	 */
-	getAccountWithdrawals(accountId) {
-		if (!isAccountId(accountId)) throw new Error('Account id is invalid');
-
-		return this.wsRpc.call([0, 'get_account_withdrawals', [accountId]]);
+	getAccountWithdrawals(idOfAccount) {
+		return this.wsRpc.call([0, 'get_account_withdrawals', [accountId.toRaw(idOfAccount)]]);
 	}
 
 	/**
 	 * @method approveProposal
 	 * @param {String} feePayingAccountId
-	 * @param {String} proposalId
+	 * @param {String} idOfProposal
 	 * @param {Object} delta
 	 * @param {Boolean} shouldDoBroadcastToNetwork
 	 * @returns {Promise<SignedTransaction>}
 	 */
-	approveProposal(feePayingAccountId, proposalId, delta, shouldDoBroadcastToNetwork) {
-		if (!isAccountId(feePayingAccountId)) {
-			throw new Error('Account id is invalid');
-		}
-		if (!isString(proposalId)) throw new Error('proposal Id should be a string');
-		if (!isObject(delta)) throw new Error('delta should be a object');
-		if (!isBoolean(shouldDoBroadcastToNetwork)) return Promise.reject(new Error('Broadcast should be a boolean'));
-
-		return this.wsRpc.call([0, 'approve_proposal',
-			[feePayingAccountId, proposalId, delta, shouldDoBroadcastToNetwork],
-		]);
+	approveProposal(feePayingAccountId, idOfProposal, delta, shouldDoBroadcastToNetwork) {
+		return this.wsRpc.call([0, 'approve_proposal', [
+			accountId.toRaw(feePayingAccountId),
+			proposalId.toRaw(idOfProposal),
+			wallet.toRaw(delta),
```

Comment:
`wallet` is collection of wallet-serializators. It has no method `toRaw`. Use `wallet.approvalDelta.toRaw`.